File: coures-2/src/systemClock.ts
import * as readlineSync from 'readline-sync';

let bootTime = new Date();
console.log("开机时间: ", bootTime.toLocaleString());

let count = 0;
let timer = 0;
let timeAlarm = readlineSync.question("请输入定时闹钟（毫秒）：");
let timeAlarmNumber = parseInt(timeAlarm, 10);

while (timeAlarmNumber > 0) {
  console.log("时钟中断，保存现场");

  timer++;
  timeAlarmNumber;

  console.log("处理器调度");

  console.log("开机时间: ", bootTime.toLocaleString(), " 定时闹钟: ", timeAlarm);

  if (++count % 10 == 0) {
    readlineSync.question("按任意键继续...");
    console.clear();
  }
}

let endTime = new Date(bootTime.getTime() + timer * 20000);
console.log("结束时间: ", endTime.toLocaleString());

File: course-2/src/systemClock.cpp
#include <iostream>
#include <iomanip>
#include <ctime>

int main() {
    time_t bootTime = time(0);
    char* s_bootTime = ctime(&bootTime);

    int count = 0;
    int timer = 0;
    int timeAlarm;

    std::cout << "请输入定时闹钟（毫秒）：";
    std::cin >> timeAlarm;

    while (timeAlarm > 0) {
        std::cout << "时钟中断，保存现场" << std::endl;

        timer++;
        timeAlarm--;

        std::cout << "处理器调度" << std::endl;

        std::cout << std::left << std::setw(16) << "开机时间: " << s_bootTime;
        std::cout << std::left << std::setw(16) << "定时闹钟: " << timeAlarm << std::endl;

        if (++count % 10 == 0) {
            std::cout << "按任意键继续..." << std::endl;
            std::cin.ignore();
            std::cin.get();
            system("clear");
        }
    }

    std::cout << std::left << std::setw(16) << "结束时间: ";
    time_t endTime = bootTime + timer * 20;
    char* s_endTime = ctime(&endTime);
    std::cout << s_endTime;

    return 0;
}
